src/main/java/nl/tudelft/ewi/devhub/server/web/resources/AssignmentsResource.java

```diff
@@ -0,0 +1,241 @@
+            throw new UnauthorizedException();
+        }
+
+        Assignment assignment = assignmentsDAO.find(course, assignmentId);
+        List<Delivery> lastDeliveries = deliveriesDAO.getLastDeliveries(assignment);
+        AssignmentStats assignmentStats = deliveriesBackend.getAssignmentStats(assignment);
+
+        Map<String, Object> parameters = Maps.newHashMap();
+        parameters.put("user", currentUser);
+        parameters.put("course", course);
+        parameters.put("assignment", assignment);
+        parameters.put("assignmentStats", assignmentStats);
+        parameters.put("lastDeliveries", lastDeliveries);
+
+        List<Locale> locales = Collections.list(request.getLocales());
+        return display(templateEngine.process("courses/assignments/assignment-view.ftl", locales, parameters));
+    }
+
+    /**
+     * An edit page page for an assignment
+     * @param request the current HttpServletRequest
+     * @param courseCode the course to create an assignment for
+     * @param assignmentId the assignment id
+     */
+    @GET
+    @Transactional
+    @Path("{assignmentId : \\d+}/edit")
+    public Response getEditAssignmentPage(@Context HttpServletRequest request,
+                                          @PathParam("courseCode") String courseCode,
+                                          @PathParam("assignmentId") Long assignmentId,
+                                          @QueryParam("error") String error) throws IOException {
+
+        if(!(currentUser.isAdmin())) {
+            throw new UnauthorizedException();
+        }
+
+        Course course = courses.find(courseCode);
+        Assignment assignment = assignmentsDAO.find(course, assignmentId);
+
+        Map<String, Object> parameters = Maps.newHashMap();
+        parameters.put("user", currentUser);
+        parameters.put("course", course);
+        parameters.put("assignment", assignment);
+
+        if(error != null)
+            parameters.put("error", error);
+
+        List<Locale> locales = Collections.list(request.getLocales());
+        return display(templateEngine.process("courses/assignments/create-assignment.ftl", locales, parameters));
+    }
+
+    @POST
+    @Path("{assignmentId : \\d+}/edit")
+    public Response editAssignment(@Context HttpServletRequest request,
+                                   @PathParam("courseCode") String courseCode,
+                                   @FormParam("id") long assignmentId,
+                                   @FormParam("name") String name,
+                                   @FormParam("summary") String summary,
+                                   @FormParam("due-date") String dueDate) {
+
+        Course course = courses.find(courseCode);
+
+        if(!(currentUser.isAdmin() || currentUser.isAssisting(course))) {
+            throw new UnauthorizedException();
+        }
+
+        Assignment assignment = assignmentsDAO.find(course, assignmentId);
```

Comment:
We can also just use `@Pathparam` for `assignmentId`. This way we don't take the assignment Id from the form into account, but this doesn't matter, is disabled anyways. Assignments Id's can't change. Either way, the problem you found (being able to edit another assignment) will be fixed.